### PR TITLE
Tractogram scripts part 5

### PR DESCRIPTION
**Pull request by EmmaRenauld**
# Quick description

Next step in tractogram scripts verification. All the next scripts were already pretty much ok! Yeah! Pretty much only formatting changes, and more verifications!

Stopped after 6 scripts. A part 6 will follow :)

- `scil_tractogram_math`
- `scil_tractogram_pairwise_comparison`
- `scil_tractogram_project_maps_to_streamlines`
- `scil_tractogram_project_streamlines_to_map`
- `scil_tractogram_qbx`
- `scil_tractogram_register`

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scilpy/tractograms/tractogram_operations.py [hunk 1]
```diff
@@ -201,8 +202,7 @@ def perform_tractogram_operation_on_sft(op_name, sft_list, precision,
         The final SFT
     """
     # Performing operation
```
Comment (CHrlS98) [inline scilpy/tractograms/tractogram_operations.py:204]: I know you didn't change this line but that comment isn't particularly useful no?

File: scripts/scil_tractogram_math.py [hunk 2]
```diff
@@ -113,22 +112,24 @@ def main():
...
     logging.getLogger().setLevel(logging.getLevelName(args.verbose))
 
+    # Verifications
     assert_inputs_exist(parser, args.in_tractograms, args.reference)
     assert_outputs_exist(parser, args, args.out_tractogram,
                          optional=args.save_indices)
     assert_headers_compatible(parser, args.in_tractograms,
                               reference=args.reference)
 
+    # Lazy: Loading and processing are done on-the-fly. Then exit script.
```
Comment (CHrlS98) [inline scripts/scil_tractogram_math.py:122]: Would put the comment `#Then exit script` above the `return`.

File: scripts/scil_tractogram_math.py [hunk 3]
```diff
@@ -137,24 +138,27 @@ def main():
                              header=header)
         return
 
-    # Load all input streamlines.
+    # Non-lazy:
```
Comment (CHrlS98) [inline scripts/scil_tractogram_math.py:141]: a bit more explicit? `#All non-lazy operations are handled below`?

File: scripts/scil_tractogram_math.py [hunk 4]
```diff
@@ -113,22 +112,24 @@ def main():
     args = parser.parse_args()
     logging.getLogger().setLevel(logging.getLevelName(args.verbose))
 
+    # Verifications
```
Comment (CHrlS98) [inline scripts/scil_tractogram_math.py:115]: Do we really need this comment? the methods are already pretty explicit in what they do...

Comment (EmmaRenauld) [reply scripts/scil_tractogram_math.py:115]: So far, all scripts I checked have been separated into: verifications, loading, processing, saving. Easier to proof-read for me.

**Comment by frheault** [comment]
@EmmaRenauld I just realized that scil_connectivity_pairwise_agreement.py use the word agreement, not comparison. Should it be a small PR or should we wait for a big connectivity refactor?

**Comment by EmmaRenauld** [comment]
> I just realized that scil_connectivity_pairwise_agreement.py use the word agreement, not comparison. Should it be a small PR or should we wait for a big connectivity refactor?

Indeed, I see that we have 
-scil_tractogram_pairwise_comparison
-scil_bundle_pairwise_comparison
-scil_connectivity_pairwaise_agreement.

Is there a fundamental difference between these 3 scripts in the reason why we would want to use them, of the type of study? 

If not, I can change it if you want, in any PR.

**Review by arnaudbore — APPROVED** [review]
LGTM !

**Review by CHrlS98 — APPROVED** [review]
Good

**Pull request merged**